core/src/avm2/globals/flash/net/socket.rs

```diff
@@ -0,0 +1,405 @@
+        }
+    };
+
+    // FIXME: Are we supposed to throw and IOError when a connection fails?
+
+    Ok(Value::Undefined)
+}
+
+pub fn close<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(socket) = this.as_socket() {
+        // We throw an IOError when socket is not open.
+        let handle = socket
+            .get_handle()
+            .ok_or(Error::AvmError(io_error(activation, "TODO", 0)?))?;
+
+        let UpdateContext { sockets, .. } = &mut activation.context;
+
+        sockets.close(handle)
+    }
+
+    Ok(Value::Undefined)
+}
+
+pub fn get_bytes_available<'gc>(
+    _activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(socket) = this.as_socket() {
+        return Ok(socket.read_buffer().len().into());
+    }
+
+    Ok(Value::Undefined)
+}
+
+pub fn get_endian<'gc>(
+    _activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(socket) = this.as_socket() {
+        return Ok(match socket.endian() {
+            Endian::Big => "bigEndian".into(),
+            Endian::Little => "littleEndian".into(),
+        });
+    }
+
+    Ok(Value::Undefined)
+}
+
+pub fn set_endian<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(socket) = this.as_socket() {
+        let endian = args
+            .get(0)
+            .unwrap_or(&Value::Undefined)
+            .coerce_to_string(activation)?;
+        if &endian == b"bigEndian" {
+            socket.set_endian(Endian::Big);
+        } else if &endian == b"littleEndian" {
+            socket.set_endian(Endian::Little);
+        } else {
+            return Err(make_error_2008(activation, "value"));
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+pub fn get_connected<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let socket = match this.as_socket() {
+        Some(socket) => socket,
+        None => return Ok(Value::Undefined),
+    };
+
+    let UpdateContext { sockets, .. } = &mut activation.context;
+
+    let handle = match socket.get_handle() {
+        Some(handle) => handle,
+        None => return Ok(Value::Bool(false)),
+    };
+
+    let is_connected = sockets.is_connected(handle).unwrap_or(false);
+
+    Ok(Value::Bool(is_connected))
+}
+
+pub fn flush<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    if let Some(socket) = this.as_socket() {
+        let UpdateContext { sockets, .. } = &mut activation.context;
+
+        // FIXME: Throw correct IoError
+        let handle = socket.get_handle().unwrap();
```

Comment:
`io_error(activation, "Error #2002: Operation attempted on invalid socket.", 2002)`
Seeing as this error is going to appear often, it might be a good idea to create a function like:
```
fn invalid_socket_error<'gc>(
    activation: &mut Activation<'_, 'gc>,
) -> Error<'gc> {
    io_error(activation, "Error #2002: Operation attempted on invalid socket.", 2002)
}
```